sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -188,45 +179,52 @@ private static void WriteMemory(MemoryStream stream, ReadOnlyMemory<byte> memory
-        public virtual Response<UploadLogsResult> Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
+        public virtual Response Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(ruleId, nameof(ruleId));
             Argument.AssertNotNullOrEmpty(streamName, nameof(streamName));
             Argument.AssertNotNullOrEmpty(logs, nameof(logs));
 
             using var scope = ClientDiagnostics.CreateScope("LogsIngestionClient.Upload");
 
-            RequestContext requestContext = GenerateRequestContext(cancellationToken);
             Response response = null;
-            List<UploadLogsError> errors = new List<UploadLogsError>();
+            var exceptions = new List<Exception>();
+
             scope.Start();
-            try
+            // Keep track of the number of failed logs across batches
+            int logsFailed = 0;
+
+            // Partition the stream into individual blocks
+            foreach (BatchedLogs<T> batch in Batch(logs, options))
             {
-                // Partition the stream into individual blocks
-                foreach (BatchedLogs<T> batch in Batch(logs, options))
+                try
                 {
                     // Because we are uploading in sequence, wait for each batch to upload before starting the next batch
-                    BatchUpload task = CommitBatchListSyncOrAsync(
+                    response = UploadBatchListSyncOrAsync(
                         batch,
                         ruleId,
                         streamName,
-                        false,
+                        async: false,
                         cancellationToken).EnsureCompleted();
 
-                    // If we have an error, add error from response into errors list which represents the errors from all the batches
-                    if (task.Error != null)
-                        errors.Add(task.Error);
+                    if (response.Status != 204)
+                    {
+                        throw new RequestFailedException(response);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    logsFailed += batch.LogsList.Count;
+                    // If we have an error, add error from response into exceptions list which represents the errors from all the batches
+                    exceptions = AddException(
+                    exceptions,
+                    ex);
                 }
             }
-            catch (Exception ex)
-            {
-                scope.Failed(ex);
-                throw;
-            }
+            ThrowException(logs, scope, exceptions, logsFailed);
```

Comment:
I think this method has enough parameters that it is probably more confusing than helpful. I would just repeat the code in the sync/async methods.